rustual-boy-core/src/rom.rs

```diff
@@ -22,19 +22,22 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        Rom::from_bytes(vec.into_boxed_slice())
+        Rom::from_bytes(&vec)
     }
 
-    pub fn from_bytes(mut bytes: Box<[u8]>) -> io::Result<Rom> {
-        let size = bytes.len();
+    pub fn from_bytes(bytes: &[u8]) -> io::Result<Rom> {
+        let bytes_copy = bytes.to_vec();
```

Comment:
nit: since `bytes_copy` and `bytes_box` later all basically refer to the same collection of bytes (the bytes we own) and really only differ in packaging, I think it would have been _slightly_ more clear to just use the `bytes` name again for those, but this is fine as-is :)